src/rgw/rgw_sal_motr.cc

```diff
@@ -2110,11 +2062,37 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       << "bucket/obj = " << tenant_bkt_name << "/" << source->get_name()
       << ", rc = " << rc << dendl;
 
-  //result.delete_marker = parent_op.result.delete_marker;
-  //result.version_id = parent_op.result.version_id;
   return 0;
 }
 
+int MotrObject::remove_mobj_and_index_entry(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string delete_key)
+{
+  int rc;
+  bufferlist bl;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
```

Comment:
I suppose this gets calculated/formed in the caller function. Can't we pass the same to this function ?

Reply:
bucket name not available in all currently calling functions, like @ L3292 in overwrite_null_obj() and same way not necessarily in any future use case also. And anyways such calls shouldn't be heavy, being used at quite many places